cloud/passportauth/src/user/UserSec.ts

```diff
@@ -0,0 +1,42 @@
+import BaseUser from './BaseUser';
+import UserApiService from './UserApi';
+
+export interface UserSec {
+  getLogin(id: string): void;
+  // getPassword(id: string): string;
+  comparePassword(id: string, password: string): void;
+  // getRoles(id: string): any;
+  getProfileData(id: string): void;
+}
+
+class UserSecService<T extends BaseUser> implements UserSec {
+  protected userApi: UserApiService<T>;
+
+  constructor(protected readonly userApiService: UserApiService<T>) {
+    this.userApi = userApiService;
+  }
+
+  public getLogin(id: string) {
+    return this.userApi.read(id).then((user) => {
+      return user.username;
```

Comment:
In the `/auth/PassportSetup.ts` file, there is `this.userSec.getLogin(username).then((user) => { ......` which calls the function here. A username gets passed in during auth, and then the username gets returned from the getLogin() method here? Should this be `userId` maybe?